### PR TITLE
refactor: extract triggerTopicDigestGenerationCore to drop double-auth (#452)

### PR DESCRIPTION
**Pull request by funny-otter**
## Summary

- `triggerTopicDigestRefresh` previously delegated to the public `triggerTopicDigestGeneration`, so each refresh request ran `withAuthAction` (Clerk `auth()`) twice and `topicDigestSchema.safeParse` twice. Cosmetic perf issue today, but a footgun: any future divergence between the two outer wrappers would silently leak into one path and not the other.
- Extract a private `triggerTopicDigestGenerationCore(input)` containing the shared body (canonical lookup, eligibility/staleness check, `tasks.trigger` call). Both public actions now wrap the core with a single `withAuthAction` + single `safeParse`. No wire-shape changes; no signature changes.
- Add `expect(mockAuth).toHaveBeenCalledTimes(1)` across every branch of `triggerTopicDigestGeneration` (5 tests) and `triggerTopicDigestRefresh` (5 tests) to lock in the single-auth contract as a regression guard. Hoist the rationale into one suite-level comment per `describe` block instead of repeating it inline.

## Test plan

- [x] `bun run test -- src/app/actions/__tests__/topics-digest.test.ts src/app/actions/__tests__/topics-detail.test.ts` — 36 / 36 pass
- [x] `bun run test` — full suite (3335 / 3335 pass, 56 integration tests skipped without `DATABASE_URL`; none exercise the refactored functions)
- [x] `bun run format:check` clean
- [x] `bun run lint` clean
- [x] `bun run build` succeeds
- [x] `bun run build-storybook` succeeds
- [x] Multi-tool review (Codex + pr-review-toolkit + /simplify lanes) applied: rename underscore-prefixed helper, fix misleading log tag, drop stale propagation comment, extend single-auth assertion to remaining error/edge tests

Closes #452

## Summary by CodeRabbit

* **Tests**
  * Strengthened test coverage to enforce a single authentication call across topic digest generation and refresh scenarios.

* **Refactor**
  * Internal extraction and reorganization of digest gating/trigger logic with no changes to user-facing behavior.

[Review Change Stack](https://app.coderabbit.ai/change-stack/Chalet-Labs/contentgenie/pull/465)

[Review Change Stack](https://app.coderabbit.ai/change-stack/Chalet-Labs/contentgenie/pull/465)

### CODE REVIEW

File: src/app/actions/__tests__/topics-digest.test.ts [hunk 1]
```diff
@@ -134,6 +134,7 @@ describe("triggerTopicDigestGeneration", () => {
     expect(result).toEqual({ success: false, error: "Unauthorized" });
     expect(mockDbSelect).not.toHaveBeenCalled();
     expect(mockTasksTrigger).not.toHaveBeenCalled();
+    expect(mockAuth).toHaveBeenCalledTimes(1);
   });
```
[the remark this answers is not in the document]
Comment (funny-otter) [reply src/app/actions/__tests__/topics-digest.test.ts:138]: Fixed: added `expect(mockAuth).toHaveBeenCalledTimes(1)` to the remaining `triggerTopicDigestGeneration` branches in `topics-digest.test.ts` — invalid input (case 2), unknown extra key (case 2b), canonical not found (case 3), non-active canonical (case 4), ineligible (case 5), and cached (case 6). The single-`auth()` contract is now asserted across all 10 cases in that describe block.

File: src/app/actions/__tests__/topics-detail.test.ts [hunk 2]
```diff
@@ -553,6 +553,9 @@ describe("getTopicDetailData", () => {
 // ============================================================================
 
 describe("triggerTopicDigestRefresh", () => {
+  // Auth-call-count assertions across these tests guard the single-`auth()`
+  // contract from issue #452 — refresh previously delegated through the
+  // public `triggerTopicDigestGeneration`, double-wrapping `withAuthAction`.
   it("returns Unauthorized without a session", async () => {
```
[the remark this answers is not in the document]
Comment (funny-otter) [reply src/app/actions/__tests__/topics-detail.test.ts:559]: Fixed: added `expect(mockAuth).toHaveBeenCalledTimes(1)` to the remaining `triggerTopicDigestRefresh` branches — Unauthorized, invalid-canonicalTopicId boundary cases (`it.each([0, -1, 1.5])`), and the two staleness-threshold boundary tests (cached-below-threshold, queues-at-threshold). The single-`auth()` contract is now asserted across every test in the describe.
